sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlSuite.scala

```diff
@@ -1209,10 +1209,9 @@ class XmlSuite extends QueryTest with SharedSparkSession {
       "<ROW><year>2015</year><make>Chevy</make><model>Volt</model><comment>No</comment></ROW>")
     val xmlRDD = spark.sparkContext.parallelize(data)
     val ds = spark.createDataset(xmlRDD)(Encoders.STRING)
-    assert(spark.read.option("rowTag", "ROW").xml(ds).collect().length === 3)
```

Comment:
Wait, we still need this to be required I guess? `ROW` can be different in the text Dataset?

Reply:
It is [assumed](https://github.com/apache/spark/blob/8fd915ffaba1cc99813cc8d6d2a28688d7fae39b/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L576) that each XML string in the `Dataset` contains only one `Row`. So the parser doesn't look for `rowTag` as it iterates over all the strings in the Dataset.